src/data_server/utils/ranges.rs

```diff
@@ -46,8 +46,8 @@ pub fn aruna_range_from_s3range(range_string: S3Range, content_length: u64) -> A
                 to: content_length,
             },
         },
-        Suffix { last } => ArunaRange {
-            from: content_length - last,
+        Suffix { length } => ArunaRange {
```

Comment:
Have you checked that this is working ? By getting a range instead of a full file ?

Reply:
This is fixed now